main-service/src/main/java/ru/practicum/user/controller/UserController.java

```diff
@@ -0,0 +1,42 @@
+package ru.practicum.user.controller;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.*;
+import ru.practicum.user.dto.NewUserRequestDto;
+import ru.practicum.user.dto.UserDto;
+import ru.practicum.user.service.UserService;
+
+import java.util.List;
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/admin/users")
+@Validated
+public class UserController {
+
+    private final UserService userService;
+
+    @GetMapping
+    public List<UserDto> getUsers(@RequestParam(required = false) List<Long> ids,
+                                  @RequestParam(defaultValue = "0") int from,
+                                  @RequestParam(defaultValue = "10") int size) {
+        if (ids == null) {
+            return userService.getUsers(from, size);
+        }
+        return userService.getUsers(ids, from, size);
+    }
+
+    @PostMapping
+    @ResponseStatus(HttpStatus.CREATED)
+    public UserDto createUser(@RequestBody @Validated NewUserRequestDto newUserRequestDto) {
```

Comment:
Нужна аннотация Valid
```suggestion
    public UserDto createUser(@RequestBody @Valid NewUserRequestDto newUserRequestDto) {
```